commands/droplets.go

```diff
@@ -556,9 +556,35 @@ func matchDroplets(ids []string, ds do.DropletsService, fn matchDropletsFn) erro
 
 // RunDropletGet returns a droplet.
 func RunDropletGet(c *CmdConfig) error {
+
 	id, err := getDropletIDArg(c.NS, c.Args)
 	if err != nil {
-		return err
+		// Try to find Droplet by name
+		if len(c.Args) < 1 {
+			return doctl.NewMissingArgsErr(c.NS)
+		}
+		ds := c.Droplets()
+		fn := func(ids []int) error {
+			var matches []int
+			for _, id := range ids {
+				d, err := ds.Get(id)
+				if err != nil {
+					return fmt.Errorf("Unable to get Droplet %d: %v", id, err)
+				}
+				if d.Name == c.Args[0] {
+					matches = append(matches, d.ID)
+				}
+			}
+			if len(matches) == 1 {
+				id = matches[0]
+			}
+			return nil
+		}
+
+		err := matchDroplets(c.Args, ds, fn)
+		if err != nil {
+			return err
+		}
```

Comment:
Some of this is still duplicating some of what `matchDroplets` provides. The "anonymous" function here that is passed to `matchDroplets`, is called with a list of IDs which have already had the conversion from name to ID. This also duplicates the `ds.Get(id)` further below, adding another network request.

It's probably a bit cleaner to move the display logic into the function and just return the call to `matchDroplets` at the end.

```go
	fn := func(ids []int) error {
		for _, id := range ids {
			d, err := ds.Get(id)
			if err != nil {
				return err
			}

			item := &displayers.Droplet{Droplets: do.Droplets{*d}}
			if getTemplate != "" {
				t := template.New("Get template")
				t, err = t.Parse(getTemplate)
				if err != nil {
					return err
				}
				return t.Execute(c.Out, d)
			}
			return c.Display(item)
		}
		return nil
	}

	return matchDroplets(c.Args, ds, fn)
```